src/app/buy-credits/buy-credits.component.ts

```diff
@@ -0,0 +1,245 @@
+
+  ngOnInit(): void {
+
+    const idAndJWT = this.identityService.getIdAndJWT();
+    if (idAndJWT !== undefined) {
+      if (this.identityService.isTokenForFinalisedUser(idAndJWT.jwt)) {
+        this.loadAuthedPersonInfo(idAndJWT.id, idAndJWT.jwt);
+      }
+    }
+
+    else {
+      this.showLoginDialog();
+    }
+
+    const formGroups: {
+      amounts: FormGroup,
+      giftAid: FormGroup
+    } = {
+      amounts: this.formBuilder.group({
+        creditAmount: [null, [
+          Validators.required,
+          ValidateCreditMin,
+          ValidateCreditMax,
+          Validators.pattern('^[£$]?[0-9]+?(\\.00)?$'),
+        ]],
+        tipPercentage: [this.initialTipSuggestedPercentage],
+        customTipAmount: [null, [
+          // Explicitly enforce minimum custom tip amount of £0. This is already covered by the regexp
+          // validation rule below, but it's good to add the explicit check for future-proofness
+          Validators.min(0),
+          // Below we use the donation flow validator (ValidateCurrencyMax) for the tip because
+          // when buying donation credits, tips are set as real donations to a dedicated Big Give
+          // SF campaign. See MAT-266 and the Slack thread linked it its description for more context.
+          ValidateCurrencyMax,
+          Validators.pattern('^[£$]?[0-9]+?(\\.00)?$'),
+        ]],
+      }),
+      giftAid: this.formBuilder.group({
+        giftAid: [null],
+        homeAddress: [null],
+        homeBuildingNumber: [null],
+        homeOutsideUK: [null],
+        homePostcode: [null],
+      }),
+      // T&Cs agreement is implicit through submitting the form.
+    };
+
+    this.creditForm = this.formBuilder.group(formGroups);
+
+    const amountsGroup: any = this.creditForm.get('amounts');
+    if (amountsGroup != null) {
+      this.amountsGroup = amountsGroup;
+    }
+
+    const giftAidGroup: any = this.creditForm.get('giftAid');
+    if (giftAidGroup != null) {
+      this.giftAidGroup = giftAidGroup;
+    }
+  }
+
+  buyCredits(): void {
+    this.isLoading = true;
+    const idAndJWT = this.identityService.getIdAndJWT();
+    if (idAndJWT !== undefined) {
+      this.identityService.getFundingInstructions(idAndJWT?.id, idAndJWT.jwt).subscribe((response: FundingInstruction) => {
+        this.isLoading = false;
+        this.isPurchaseComplete = true;
+        this.accountNumber = response.bank_transfer.financial_addresses[0].sort_code.account_number;
+        this.sortCode = response.bank_transfer.financial_addresses[0].sort_code.sort_code;
+        this.accountHolderName = response.bank_transfer.financial_addresses[0].sort_code.account_holder_name;
+      });
+    }
+  }
+
+  giftAidChoiceSelected(e: MatRadioChange) {
+    if (e.value === "yes") {
+      this.isOptedIntoGiftAid = true;
+    }
+    else {
+      this.isOptedIntoGiftAid = false;
+    }
+  }
+
+  showLoginDialog() {
+    const loginDialog = this.dialog.open(DonationStartLoginDialogComponent);
```

Comment:
Good that we can reuse this, but I guess if doing so we should rename it + move it up in the app structure to avoid confusion